src/ServicePulse.Host/app/modules/monitoring/js/services/services.messageTypeParser.js

```diff
@@ -9,6 +9,7 @@
 
             if (messageType.typeName.indexOf(';') > 0) {
                 var messageTypeHierarchy = messageType.typeName.split(';');
+                messageTypeHierarchy = messageTypeHierarchy.replace('+', '.');
```

Comment:
isn't `messageTypeHierarchy` a array given it's assigned by `typeName.split(';')` ? Does that have a `replace` method?